### PR TITLE
feat: adds working dir support, updates versions

### PR DESCRIPTION
**Pull request by dave-gray101**
* Updated gopsutils to v4
  * bumps module target to golang 1.18+
* Allows setting working directory
* _Breaking Change_ `config.KillSignal` is now an `*int` - avoids the need to parse at runtime. `options.WithKillSignal` will need to be updated.

### CODE REVIEW

**Comment by dave-gray101** [comment]
@mudler

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2021 Ettore Di Giacinto
+Copyright (c) 2024 Ettore Di Giacinto and Dave Lee
```
Comment (mudler) [inline LICENSE:4]:
```suggestion
Copyright (c) 2021-2023 Ettore Di Giacinto
Copyright (c) 2024 Ettore Di Giacinto and Dave Lee
```

**Review by mudler — APPROVED** [review]

**Comment by mudler** [comment]
mm stinky error:

```
go: downloading golang.org/x/text v0.3.6
Error: ../../../go/pkg/mod/github.com/shirou/gopsutil/v4@v4.24.7/cpu/cpu_aix_cgo.go:9:2: C source files not allowed when not using cgo or SWIG: c_helpers.c
Error: ../../../go/pkg/mod/github.com/tklauser/numcpus@v0.6.1/numcpus_linux.go:24:2: found packages unix (affinity_linux.go) and main (mkasm.go) in /home/runner/go/pkg/mod/golang.org/x/sys@v0.22.0/unix
Error: Process completed with exit code 1.
```

**Comment by dave-gray101** [comment]
> mm stinky error:
> 
> ```
> go: downloading golang.org/x/text v0.3.6
> Error: ../../../go/pkg/mod/github.com/shirou/gopsutil/v4@v4.24.7/cpu/cpu_aix_cgo.go:9:2: C source files not allowed when not using cgo or SWIG: c_helpers.c
> Error: ../../../go/pkg/mod/github.com/tklauser/numcpus@v0.6.1/numcpus_linux.go:24:2: found packages unix (affinity_linux.go) and main (mkasm.go) in /home/runner/go/pkg/mod/golang.org/x/sys@v0.22.0/unix
> Error: Process completed with exit code 1.
> ```

Good catch - since the fork wasn't set up for tests, I missed that the yaml matrix needs to be updated to match the versions required by the updated gopsutil. Correcting this now.

**Pull request merged**